src/api/api.ts

```diff
@@ -7,141 +14,200 @@ import {
-  installedRequest(client, `api/v4/contacts`, "GET");
+  request(client, { endpoint: `api/v4/contacts`, method: "GET", settings });
 
 export const getTasksByPrompt = async (
   client: IDeskproClient,
-  prompt: string
+  prompt: string,
+  settings: RequestParams["settings"],
 ): Promise<IWrikeResponse<ITaskFromList[]>> =>
-  installedRequest(
-    client,
-    `api/v4/tasks?title=${encodeURIComponent(
+  request(client, {
+    endpoint: `api/v4/tasks?title=${encodeURIComponent(
       prompt
     )}&descendants=true&pageSize=1000`,
-    "GET"
-  );
+    method: "GET",
+    settings,
+  });
 
 export const getWorkflows = async (
-  client: IDeskproClient
+  client: IDeskproClient,
+  settings: RequestParams["settings"],
 ): Promise<IWrikeResponse<IWorkflow[]>> =>
-  installedRequest(client, `api/v4/workflows`, "GET");
+  request(client, { endpoint: `api/v4/workflows`, method: "GET", settings });
 
 export const getTasks = async (
-  client: IDeskproClient
+  client: IDeskproClient,
+  settings: RequestParams["settings"],
 ): Promise<IWrikeResponse<ITaskFromList[]>> =>
-  installedRequest(
-    client,
-    `api/v4/tasks?descendants=true&pageSize=1000`,
-    "GET"
-  );
+  request(client, {
+    endpoint: `api/v4/tasks?descendants=true&pageSize=1000`,
+    method: "GET",
+    settings,
+  });
 
 export const getTasksByIds = async (
   client: IDeskproClient,
-  ids: string[]
+  ids: string[],
+  settings: RequestParams["settings"],
 ): Promise<IWrikeResponse<ITaskFromList[]>> =>
-  installedRequest(client, `api/v4/tasks/${ids.join(",")}`, "GET");
+  request(client, {
+    endpoint: `api/v4/tasks/${ids.join(",")}`,
+    method: "GET",
+    settings,
+  });
 
 export const getTaskById = async (
   client: IDeskproClient,
-  taskId: string
+  taskId: string,
+  settings: RequestParams["settings"],
 ): Promise<IWrikeResponse<ITask[]>> =>
-  installedRequest(client, `api/v4/tasks/${taskId}`, "GET");
+  request(client, {
+    endpoint: `api/v4/tasks/${taskId}`,
+    method: "GET",
+    settings,
+  });
 
 export const getTasksByFolderId = async (
   client: IDeskproClient,
-  folderId: string
+  folderId: string,
+  settings: RequestParams["settings"],
 ): Promise<IWrikeResponse<ITask[]>> =>
-  installedRequest(client, `api/v4/folders/${folderId}/tasks`, "GET");
+  request(client, {
+    endpoint: `api/v4/folders/${folderId}/tasks`,
+    method: "GET",
+    settings,
+  });
 
-export const getFolders = async (client: IDeskproClient) => {
-  const folderData: IWrikeResponse<IFolderFromList[]> = await installedRequest(
-    client,
-    "api/v4/folders",
-    "GET"
-  );
+export const getFolders = async (
+  client: IDeskproClient,
+  settings: RequestParams["settings"],
+) => {
+  const folderData: IWrikeResponse<IFolderFromList[]> = await request(client, {
+    endpoint: "api/v4/folders",
+    method: "GET",
+    settings,
+  });
 
   return {
     data: folderData.data.filter((folder) => folder.title !== "Recycle Bin"),
```

Comment:
Why don't we want to show the recycle bin? And if so, can this be done server side?

Reply:
I have no idea why it was done this way, but I assume there must have been reasons for it. I'll try to figure it out in this story and leave a comment on why we do this
https://app.shortcut.com/deskpro/story/161047/wrike-improvements